zproject/backends.py

```diff
@@ -1547,6 +1549,30 @@ def social_auth_finish(
     else:
         is_signup = False
 
+    extra_attrs = return_data.get("extra_attrs", {})
+    if (
+        user_profile is not None
+        and extra_attrs
+        and realm.subdomain in settings.SOCIAL_AUTH_SYNC_CUSTOM_ATTRS_DICT
+    ):
+        # This is only supported for SAML right now, though the design
+        # is meant to be easy to extend this to other backends if desired.
```

Comment:
It's probably worth extending the comment to note that in contrast with LDAP, syncing custom profile fields can only happen on-login, since we don't have a way to query the social auth backend except during an authentication.

Reply:
Added. We may get a non-ldap way of querying if we implement SCIM at some point (which we've looked into in the past), but it'll be a separate system anyway, so the comment seems in order regardless.

Reply:
Sounds good.